docs/dev.md

```diff
@@ -0,0 +1,54 @@
+# Development
+
+This page is designed for software developers who are helping to develop this Lean VS Code exension.
+
+The repository contains NPM packages implementing editor integration for the Lean 4 programming language. The VS Code
+extension proper is in [`vscode-lean4`](vscode-lean4/), while [`lean4-infoview`](lean4-infoview/) implements the
+information display. The API glue code between them is in [`lean4-infoview-api`](lean4-infoview-api/). We build
+the packages in tandem using Lerna.
+
+![overview](images/overview.png)
+
+### Building
+- Make sure you have an up to date installation of `npm` and `node.js`.  For example `npm` version 8.1.3 and `node.js` version v16.13.0.
+- Run `npm install` in this folder. This installs the Lerna package manager.
+- Run `npx lerna bootstrap`. This sets up the project's dependencies.
+- Run `npx lerna run build`. This compiles the extension (which is necessary for go-to-definition in VS Code).
+
+### Testing
+- Make sure you have executed the above build steps
+- Run `npm run test`
+- See [test readme](vscode-lean4/test/readme.md) for more information.
+
+### Debugging
+- Open VS Code on this folder.
+- Press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>B</kbd> to compile the extension. This step is needed for a working development setup.
+- Press F5 to launch the extension in the VS Code debugger, (the correct launch.json configuration is already checked into this repo.)
+
+To debug the tests use the launch configuration named `Extension Tests - SuiteName`.
+
+**Note:** if breakpoints are not working, try changing one line of code in `function activate` in `extension.ts`, even
+adding a newline seems to work, then press F5.
+
+## Packaging
+
+To publish the extension on the VS Code marketplace you
+should follow [these instructions](https://code.visualstudio.com/api/working-with-extensions/publishing-extension).
```

Comment:
```suggestion
To publish the extension on the VS Code marketplace you
should run the `./release.sh` script.
```

Reply:
ok, I merged your link to the script, but included the link for more information (that link shows how to install vsce, etc)